Release 0.5.0 (#32)
nginxinc/nginx-plus-go-client: CHANGELOG.md

## 0.5.0 (Sep 25, 2019)
FEATURES:
* [30](https://github.com/nginxinc/nginx-plus-go-client/pull/30): *Additional Upstream Server Parameters*.

> The client now supports configuring `route` `backup` `down` `drain` `weight` `service` in upstream.

Reviewer: ```suggestion
The client now supports configuring `route`, `backup`, `down`, `drain`, `weight` and `service` in upstream.
```

Reviewer 2: ```suggestion
* [30](https://github.com/nginxinc/nginx-plus-go-client/pull/30): *Support additional upstream server parameters*.
The client now supports configuring `route`, `backup`, `down`, `drain`,  `weight` and `service` parameters for http upstreams and  `backup`, `down`,  `weight` and  `service` parameters for stream upstreams.
```

> * [31](https://github.com/nginxinc/nginx-plus-go-client/pull/31): *Support `location zones` and `resolver` metrics*.

Reviewer 2: ```suggestion
* [31](https://github.com/nginxinc/nginx-plus-go-client/pull/31): *Support location zones and resolver metrics*.
```

BUGFIXES:

> * [29](https://github.com/nginxinc/nginx-plus-go-client/pull/29): *Fix `MaxFails` in UpstreamServer and StreamUpstreamServer and improved test coverage*. Previously the default was `0` as opposed to the intended `1`.

Reviewer: Is test coverage improvement is relevant to changelog? 

```suggestion
* [29](https://github.com/nginxinc/nginx-plus-go-client/pull/29): *Fix `MaxFails` in UpstreamServer and StreamUpstreamServer*. Previously the default was `0` as opposed to the intended `1`.
```

Reviewer 2: ```suggestion
[29](https://github.com/nginxinc/nginx-plus-go-client/pull/29): *Fix max_fails parameter in upstream servers*. Previously, if the MaxFails field was not explicitly set, the client would incorrectly configure an upstream with the value `0` instead of the correct value `1`.
```

> 

Reviewer 2: we need to add the changes field:
```
CHANGES:
 
* The version of NGINX Plus for e2e testing was changed to R19.
* The version of the API was changed to 5.
```

## 0.5.0 (Sep 25, 2019)
FEATURES:
* [30](https://github.com/nginxinc/nginx-plus-go-client/pull/30): *Support additional upstream server parameters*.
The client now supports configuring `route`, `backup`, `down`, `drain`,  `weight` and `service` parameters for http upstreams and  `backup`, `down`,  `weight` and  `service` parameters for stream upstreams.
* [31](https://github.com/nginxinc/nginx-plus-go-client/pull/31): *Support location zones and resolver metrics*.

BUGFIXES:
* [29](https://github.com/nginxinc/nginx-plus-go-client/pull/29): *Fix max_fails parameter in upstream servers*. Previously, if the MaxFails field was not explicitly set, the client would incorrectly configure an upstream with the value `0` instead of the correct value `1`.

CHANGES:

> 

Reviewer: ```suggestion
```
